projects/go-lib/src/lib/components/go-timepicker/go-time.component.ts

```diff
@@ -0,0 +1,230 @@
+          }
+          this.autoFocousInput();
+        }
+        this.opened = value;
+      }
+    );
+  }
+
+  autoFocousInput(): void {
+    setTimeout(() => {
+      document.getElementById('hour-input').focus();
+    });
+  }
+
+  formatAMPM(date: any): void {
+    let hours: any = date.getHours();
+    let minutes: any = date.getMinutes();
+    const ampm: any = hours >= 12 ? 'pm' : 'am';
+    hours = hours % 12;
+    hours = hours ? hours : 12; // the hour '0' should be '12'
+    hours = hours > 9 ? hours : '0' + hours;
+    minutes = minutes > 9 ? minutes : '0' + minutes;
+    const formatTime: any = { hours, minutes, ampm };
+    return formatTime;
+  }
+  changeTimeFormat(timeString: any): void {
+    const H: any = +timeString.substr(0, 2);
+    const h: any = H % 12 || 12;
+    const ampm: any = H < 12 || H === 24 ? 'AM' : 'PM';
+    const minute: any = timeString.substr(3, 2);
+    this.hour = h > 9 ? h : '0' + h;
+    this.minute = (minute > 9 || minute === '00' || minute.length === 2) ? minute : '0' + minute;
+    this.format = ampm === 'AM' ? true : false;
+  }
+  changeFormat(): void {
+    this.format = !this.format;
+  }
+  increaseHour(): void {
+    if (this.hour >= 12) {
+      this.hour = '00';
+      return;
+    }
+    if (Number(this.hour) < 9) {
+      this.hour = '0' + (Number(this.hour) + 1);
+    } else {
+      this.hour = Number(this.hour) + 1;
+    }
+  }
+  increaseMinute(): void {
+    if (this.minute >= 59) {
+      this.minute = '00';
+      return;
+    }
+    if (Number(this.minute) < 9) {
+      this.minute = '0' + (Number(this.minute) + 1);
+    } else {
+      this.minute = Number(this.minute) + 1;
+    }
+  }
+  decreaseHour(): void {
+    if (this.hour === '00') {
+      this.hour = '00';
+      return;
+    }
+    if (Number(this.hour) <= 10) {
+      this.hour = '0' + (Number(this.hour) - 1);
+    } else {
+      this.hour = Number(this.hour) - 1;
+    }
+  }
+  decreaseMinute(): void {
+    if (this.minute === '00') {
+      this.minute = '00';
+      return;
+    }
+    if (Number(this.minute) <= 10) {
+      this.minute = '0' + (Number(this.minute) - 1);
+    } else {
+      this.minute = Number(this.minute) - 1;
+    }
+  }
+
+  twoDigitNumber(e) {
+    e = e || window.event;
+    const charCode = typeof e.which === 'number' ? e.which : e.keyCode;
+    if (e.target.value.length >= 2) {
+      e.target.value = '';
+      // return false;
+    }
+    // Allow non-printable keys
+    if (!charCode || charCode === 8 /* Backspace */) {
+      return;
+    }
+
+    const typedChar = String.fromCharCode(charCode);
+
+    // Allow numeric characters
+    if (/\d/.test(typedChar)) {
+      return;
+    }
+
+    // Allow the minus sign (-) if the user enters it first
+    if (typedChar === '-' && this.hour === '') {
+      return;
+    }
+
+    // In all other cases, suppress the event
+    return false;
+  }
+  clear(): void {
+    this.clickActionButton = true;
+    this.resetTimeInput();
```

Comment:
These lines should be able to be removed if we add `*ngIf="goTime.isOpen"` in the go-timepicker html